proposals/repository.go

```diff
@@ -72,7 +75,39 @@ func (r *Repository) fetch(ctx context.Context, status string) ([]*models.GetPro
 	return resp.GetPayload().Proposals, nil
 }
 
-func (r *Repository) GetVotes(ctx context.Context, proposalID int) ([]*models.GetProposalVotesResultVotes, error) {
+func (r *Repository) GetVotes(ctx context.Context, proposalID int) ([]*query.VotesOKBodyVotesItems0, error) {
+	var paginationKey strfmt.Base64
+	limit := "1000"
+	votes := make([]*query.VotesOKBodyVotesItems0, 0)
+	for {
+		resp, err := r.apiClient.Query.Votes(
+			&query.VotesParams{
+				PaginationKey:   &paginationKey,
+				PaginationLimit: &limit,
+				ProposalID:      strconv.Itoa(proposalID),
+				Context:         ctx,
+			},
+		)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get votes for proposal id = %d: %w", proposalID, err)
+		}
+
+		err = resp.GetPayload().Validate(nil)
```

Comment:
is `resp.GetPayload()` parsing something or is it just a getter for a pre-parsed object? if it's parsing something we might want to do it just once

Reply:
just a getter to a struct field